api/v1beta1/remotesecret_types.go

```diff
@@ -82,6 +105,92 @@ type TargetStatus struct {
 	Error string `json:"error,omitempty"`
 }
 
+// TargetKey is not used in the RemoteSecret spec as such but it represents an identifier of a target. As such it can be used
+// as key in the maps instead of RemoteSecretTarget (which itself is not comparable).
+//
+// Note that it is not directly possible to compare a TargetKey generated from a RemoteSecretTarget and TargetStatus, because
+// the spec may contain an empty secret name and non-empty generate name. To determine whether a target key may correspond to
+// another, use the CorrespondsTo function on the target key.
+type TargetKey struct {
+	ApiUrl             string
+	Namespace          string
+	SecretName         string
+	SecretGenerateName string
+}
+
+func (ts TargetStatus) ToTargetKey() TargetKey {
+	return TargetKey{
+		ApiUrl:             ts.ApiUrl,
+		Namespace:          ts.Namespace,
+		SecretName:         ts.SecretName,
+		SecretGenerateName: "",
+	}
+}
+
+// ToTargetKey converts the remote secret target into a target key given the default spec
+// specified in the provided remote secret.
+func (rst RemoteSecretTarget) ToTargetKey(containingRemoteSecret *RemoteSecret) TargetKey {
+	secretName := rst.Secret.Name
+	if secretName == "" {
+		secretName = containingRemoteSecret.Spec.Secret.Name
+	}
+	secretGenerateName := rst.Secret.GenerateName
+	if secretGenerateName == "" {
+		secretGenerateName = containingRemoteSecret.Spec.Secret.GenerateName
+	}
+	return TargetKey{
+		ApiUrl:             rst.ApiUrl,
+		Namespace:          rst.Namespace,
+		SecretName:         secretName,
+		SecretGenerateName: secretGenerateName,
+	}
+}
+
+// Correspondence is a result of the TargetKey#CorrespondsTo method describing how two targets correspond to each
+// other. Valid values are NameCorrespondence, GenerateNameCorrespondence and NoCorrespondence.
+type Correspondence int
+
+const (
+	NoCorrespondence Correspondence = iota
+	GenerateNameCorrespondence
+	NameCorrespondence
+)
+
+// CorrespondsTo tells whether the target key corresponds to the other target key. Note that this
+// operation is not reflective, i.e. a.CorrespondsTo(b) does not imply b.CorrespondsTo(a).
```

Comment:
*symetric :)

Reply:
It's been a looong time since I last used these words 🙃